sti-common-util/src/main/java/cz/cuni/mff/xrg/odalic/util/parsing/UriParsingUtil.java

```diff
@@ -0,0 +1,60 @@
+package cz.cuni.mff.xrg.odalic.util.parsing;
+
+import uk.ac.shef.dcs.util.StringUtils;
+
+public class UriParsingUtil {
+
+    public static String parseLabelFromResourceUri(String resourceURI, boolean applyCustomHeuristics) {
+        if (!resourceURI.startsWith("http")) {
+            return resourceURI;
+        }
+
+        // URI like https://www.w3.org/1999/02/22-rdf-syntax-ns#type
+        int trimPosition = resourceURI.lastIndexOf("#");
+
+        // URI like http://dbpedia.org/property/name
+        if (trimPosition == -1) {
+            trimPosition = resourceURI.lastIndexOf("/");
+        }
+
+        if (trimPosition != -1) {
+            // Remove anything that is not a character or digit
+            // TODO: For a future improvement, take into account the "_" character.
+            String stringValue = resourceURI.substring(trimPosition + 1).replaceAll("[^a-zA-Z0-9]", "").trim();
```

Comment:
This is a bit suspicious - to have custom code parsing URIs in this way. For which part do we need such URI handling?

Reply:
This is actually an existing code from the `sti-kbproxy/src/main/java/uk/ac/shef/dcs/kbproxy/sparql/SparqlProxyCore.java` class, but since I use it on more places, I extracted it to the `UriParsingUtil` class..